src/main/java/mate/academy/dao/impl/CarDaoImplJdbc.java

```diff
@@ -0,0 +1,206 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get car by ID "
+                    + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, manufacturers.name, "
+                + "manufacturers.country FROM cars "
+                + "LEFT JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE taxi_service.cars.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseDataFromResultSet(resultSet));
+            }
+            setCars(cars, connection);
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get all cars "
+                    + cars, e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, "
+                + "manufacturer_id = ? WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            removeDrivers(car, connection);
+            insertDrivers(car, connection);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = TRUE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't delete car by ID "
+                    + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT cars.id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name, manufacturers.country "
+                + "FROM cars "
+                + "JOIN cars_drivers ON cars.id = cars_drivers.car_id "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
```

Comment:
let's join `drivers` as well and check that the driver is not deleted

Reply:
added